topology/src/main/java/io/camunda/zeebe/topology/changes/TopologyChangeCoordinatorImpl.java

```diff
@@ -115,6 +140,26 @@ private void applyOperationsOnTopology(
         });
   }
 
+  private void simulateOperationsOnTopology(
+      final ClusterTopology currentClusterTopology,
+      final List<TopologyChangeOperation> operations,
+      final ActorFuture<TopologyChangeResult> future) {
+    validateTopologyChangeRequest(currentClusterTopology, operations)
```

Comment:
:x: I think this fails if `operations` is empty. In `apply` we check it and complete the future successfully.

Reply:
Fixed by reusing more code between apply and dry-run: https://github.com/camunda/zeebe/pull/15170/commits/33bbd719278f93f0488de9becb09692ea2bc12e5